### PR TITLE
Alcohol Intoxication Overhaul

### PR DESCRIPTION
**Pull request by Crosarius**
This PR will make broad changes to the way that alcohol/butanol intoxication works in game, with the intention of making alcohol consumption actually impactful, and last much longer than it currently does. It also includes some quality of life changes to the way that liver damage works, better indication of the player's level of intoxication to better communicate when the player is getting **too** drunk, and the frequency of vomiting. Players will **no longer** die extremely fast from getting drunk, will be able to get drunk, and stay drunk.

- The rate of liver alcohol filtration has been reduced to 0.015 intoxication every second. This calculates out to the elimination of 0.015 BAC (One standard drink) every ten minutes. 
- New messages have been added to indicate when the player is drunker than drunk.
               - At 0.10 BAC the "You feel drunk!" message is replaced with the message "You feel absolutely smashed!"
               - At 0.15 BAC the "You feel drunk!" message is replaced with the message "You feel the room spinning..."
- Liver damage no longer occurs when the player vomits from intoxication, and instead occurs when the player blacks out or loses consciousness (0.20 and 0.30 BAC, respectively)
- The liver will no longer regenerate BAC is above 0.20.
- The coefficient multiplier that Unathi metabolise butanol has been buffed from 3 to 9 (They now metabolise butanol at roughly the same rate that Humans metabolise Alcohol)

One thing of note is that these changes have indirectly fixed a "bug" that was occurring for Unathi with Butanolic beverages; The liver was filtering out intoxication faster than Unathi were able to metabolise some lower strength butanolic drinks, making them unable to get drunk from things like sarezhi wine. With the buff to intoxication, and the reduction in liver filtration speed, butanol is going to be returned to being very potent for Unathi (This seems to be the way it was originally intended?) so Unathi players who are used to Butanol being quite useless at intoxicating Unathi are going to need to be especially careful.

Also, I have done testing on my local test server, but to be frank the tickrate of my test server is very fast, and I'm not entirely sure just how well these changes will translate to a live environment with a heavier server load and longer tick times.

### CODE REVIEW

**Comment by Crosarius** [comment]
I definitely want to get a LOT of feedback on this, because this would be a HUGE change, and if it isn't properly telegraphed to people, might result in a lot of people who are used to sculling large amounts of whiskey like beer suddenly getting pretty damn drunk for a long time (They shouldn't die though.)

File: code/modules/mob/living/carbon/human/intoxication.dm [hunk 1]
```diff
@@ -35,14 +35,16 @@
...
 		slurring = max(slurring, 25)
 		if (prob(5))
 			to_chat(src,"<span class='notice'>You feel drunk!</span>")
 		move_delay_mod += 2
 		sprint_cost_factor += 0.2
 
-	if(bac > INTOX_REACTION*SR)
+	if(bac > INTOX_REACTION*SR && bac < INTOX_BALANCE*SR)
+		if (prob(5))
+			to_chat(src,"<span class='notice'>You feel absolutely smashed!</span>")
```
Comment (Generalcamo) [inline code/modules/mob/living/carbon/human/intoxication.dm:47]:
```suggestion
			to_chat(src, SPAN_NOTICE("You feel absolutely smashed!"))
```

File: code/modules/mob/living/carbon/human/intoxication.dm [hunk 2]
```diff
@@ -54,10 +56,11 @@
 			var/chance = BASE_VOMIT_CHANCE + ((bac - INTOX_VOMIT*SR)*VOMIT_CHANCE_SCALE*100)
 			if (prob(chance))
 				delayed_vomit()
-				add_chemical_effect(CE_HEPATOTOXIC, 1)
 
 	if(bac > INTOX_BALANCE*SR)
 		slurring = max(slurring, 100)
+		if (prob(5))
+			to_chat(src,"<span class='notice'>You feel the room spinning...</span>")
```
Comment (Generalcamo) [inline code/modules/mob/living/carbon/human/intoxication.dm:63]:
```suggestion
			to_chat(src, SPAN_NOTICE("You feel the room spinning..."))
```

**Review by Generalcamo — CHANGES_REQUESTED** [review]

**Comment by Crosarius** [comment]
!review

**Comment by Crosarius** [comment]
With napkin math confirmed by testing, the increased intoxication from drinks is going to result in 30u of normal beer increasing BAC by 0.015. It will take 8 or 9 full glasses of beer to reach the vomiting stage. For things like whiskey, this will be about 6 shots (30 units), and for Sarezhi Wine (Unathi), they will get absolutely smashed from 30u of Sarezhi Wine.

**Review by Geevies — APPROVED** [review]

**Pull request merged**